lib/dma.c

```diff
@@ -155,20 +134,26 @@ dma_controller_remove_region(dma_controller_t *dma,
     assert(dma != NULL);
 
     for (idx = 0; idx < dma->nregions; idx++) {
+        const struct iovec *pvaddr;
         region = &dma->regions[idx];
-        if (region->dma_addr != dma_addr || region->size != size) {
+        if (region->iova.iov_base != dma_addr || region->iova.iov_len != size) {
             continue;
         }
-        err = unmap_dma(data, region->dma_addr, region->size);
+        if (region->vaddr.iov_base == MAP_FAILED) {
+            pvaddr = NULL;
+        } else {
+            pvaddr = &region->vaddr;
+        }
+        err = unmap_dma(data, &region->iova, pvaddr);
         if (err != 0) {
             vfu_log(dma->vfu_ctx, LOG_ERR,
                    "failed to notify of removal of DMA region %#lx-%#lx: %s\n",
-                   region->dma_addr, region->dma_addr + region->size,
+                   region->iova.iov_base, region->iova.iov_base + region->iova.iov_len,
```

Comment:
optional, but is it worth having

#define iov_end(iov) ((iov)->iov_base + (iov)->iov_len)

Reply:
Agreed.